Fix memory leak

pvollan opened the pull request:
#### 02309bba8c6596ade177a5c4e4b55d9dd9422966
<pre>
Fix memory leak
<a href="https://bugs.webkit.org/show_bug.cgi?id=242259">https://bugs.webkit.org/show_bug.cgi?id=242259</a>
&lt;rdar://96029757&gt;

Reviewed by Chris Dumez.

Fix memory leak after calling AudioComponentFetchServerRegistrations.

* Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
(WebKit::AuxiliaryProcessProxy::fetchAudioComponentServerRegistrations):

Canonical link: <a href="https://commits.webkit.org/252064@main">https://commits.webkit.org/252064@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
```diff
@@ -49,7 +49,9 @@
-    return WebCore::SharedBuffer::create(registrations);
+    auto audioRegistrations = adoptCF(registrations);
+    
+    return WebCore::SharedBuffer::create(audioRegistrations.get());
```
Indeed, the AudioComponentFetchServerRegistrations() doc says the caller needs to release the registrations :(

I'd rather we fix it this way though:
return WebCore::SharedBuffer::create(adoptCF(audioRegistrations).get());

pvollan replied:
> Indeed, the AudioComponentFetchServerRegistrations() doc says the caller needs to release the registrations :(
> 
> I'd rather we fix it this way though: return WebCore::SharedBuffer::create(adoptCF(audioRegistrations).get());

Ah, yes, good point! Fixed in latest patch.

Thanks for reviewing!

cdumez reviewed (approved):
r=me with improvement suggestion.

geoffreygaren commented on Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
```diff
@@ -49,7 +49,7 @@
         return nullptr;
 
-    return WebCore::SharedBuffer::create(registrations);
+    return WebCore::SharedBuffer::create(adoptCF(registrations).get());
```
r=me

pvollan committed:
Fix memory leak
https://bugs.webkit.org/show_bug.cgi?id=242259
<rdar://96029757>

Reviewed by Chris Dumez.

Fix memory leak after calling AudioComponentFetchServerRegistrations.

* Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
(WebKit::AuxiliaryProcessProxy::fetchAudioComponentServerRegistrations):

Canonical link: https://commits.webkit.org/252064@main
